generator/src/main/java/com/scottlogic/deg/generator/inputs/IndividualRuleProfileViolator.java

```diff
@@ -0,0 +1,76 @@
+package com.scottlogic.deg.generator.inputs;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.Rule;
+import com.scottlogic.deg.generator.outputs.manifest.ManifestWriter;
+import com.scottlogic.deg.generator.violations.ViolatedProfile;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Defines a profile violator which violates by violating each ruleInformation independently.
+ * Within each violated ruleInformation we violate each constraint independently. This is consistent with the current
+ * implementation of violation.
+ */
+public class IndividualRuleProfileViolator implements ProfileViolator {
+
+    private final ManifestWriter manifestWriter;
+    private final Path outputPath;
+    private final RuleViolator ruleViolator;
+
+    @Inject
+    public IndividualRuleProfileViolator(ManifestWriter manifestWriter,
+                                          @Named("outputPath") Path outputPath,
+                                          RuleViolator ruleViolator) {
+        this.manifestWriter = manifestWriter;
+        this.outputPath = outputPath;
+        this.ruleViolator = ruleViolator;
+    }
+
+    @Override
+    public List<Profile> violate(Profile profile) {
+        ArrayList<ViolatedProfile> violatedProfiles = new ArrayList<>();
+
+        // Foreach rule in the profile generate a profile with this one rule violated
+        for (Rule rule : profile.rules) {
+            violatedProfiles.add(violateRuleOnProfile(profile, rule));
+        }
+
+        try {
+            manifestWriter.writeManifest(violatedProfiles, outputPath);
+        }
+        catch (Exception e){
+            System.out.println("Failed to write manifest for violated profiles, continuing...");
```

Comment:
Shouldn't print this information to `System.out`. At worst it should go to `System.err`, but ideally rethrow the exception out.

What did this do before in this case?

Reply:
Hilariously it used to do absolutely nothing, I can switch to System.err if that fits better. This is behaviour I decided not to test.